teflo_datarouter_plugin/helpers.py

```diff
@@ -101,4 +94,58 @@ def send_post_req(dr_token_url, body):
         data = res.json()
         return data
     else:
-        raise TefloReportError(f'Generated access token Failed with status code {res["status_code"]}.')
+        raise TefloReportError(f'Generated access token Failed with status code {res.status_code}.')
+
+
+def get_req_status(get_url, ac_token, creds):
+    headers = {'Authorization': f'Bearer {ac_token}',
+               'X-DataRouter-Auth': 'openid-connect-client-credentials-grant',
+               }
+    req_count = 0
+    res = requests.get(url=get_url,
+                       headers=headers, verify=False)
+    if res.status_code == 401:
+        LOG.info('Unauthorized: Signature has expired, Creating new token and trying again(1/2).')
+        dr_id = creds['dr_client_id']
+        dr_secret = creds['dr_client_secret']
+        dr_token_url = creds['auth_url']
+
+        gen_ntoken = get_oauth_token(dr_token_url, dr_id, dr_secret)
+        headers = {'Authorization': f'Bearer {gen_ntoken}',
+                   'X-DataRouter-Auth': 'openid-connect-client-credentials-grant',
+                   }
+        res = requests.get(url=get_url,
+                           headers=headers, verify=False)
+
+    if res.status_code == 200:
+        json_content = json.loads(res.content.decode('utf8'))
+        if json_content["status"] == 'PENDING':
+            while json_content["status"] == 'PENDING' and req_count <= 10:
+                LOG.debug(f"Req status is 'PENDING'. sending req in 30 sec. attempt - {req_count + 1}/10")
+                time.sleep(30)
+                res = requests.get(url=get_url,
+                                   headers=headers, verify=False)
+                json_content = json.loads(res.content.decode('utf8'))
+                req_count = req_count + 1
+
+        if json_content["status"] == 'OK':
+            return json_content
+
+        if json_content["status"] == 'FAILURE':
+            LOG.info(f'req with id {json_content["request_id"]} failed')
+            return json_content
+
+    if res.status_code == 401:
+        LOG.info('Unauthorized: Signature has expired, Creating new token and trying again(2/2).')
+        dr_id = creds['dr_client_id']
```

Comment:
refactor this operation as you are calling the same lines of code again and again